src/ripple/protocol/Sign.h

```diff
@@ -28,23 +28,34 @@
 
 namespace ripple {
 
-/** Sign a STObject using any secret key.
-    The signature is placed in sfSignature. If
-    a signature already exists, it is overwritten.
+/** Sign an STObject
+
+    @param st Object to sign
+    @param prefix Prefix to insert before serialized object when hashing
+    @param type Signing key type used to derive public key
+    @param sk Signing secret key
+    @param sigField Field in which to store the signature on the object.
+    If not specified the value defaults to `sfSignature`.
+
+    @note If a signature already exists, it is overwritten.
 */
 void
-sign (STObject& st,
-    HashPrefix const& prefix,
-        KeyType type, SecretKey const& sk);
+sign (STObject& st, HashPrefix const& prefix,
+    KeyType type, SecretKey const& sk,
+        SF_Blob const& sigField = sfSignature);
+
+/** Returns `true` if STObject contains valid signature
 
-/** Verify the signature on a STObject.
-    The signature must be contained in sfSignature.
+    @param st Signed object
+    @param prefix Prefix inserted before serialized object when hashing
+    @param pk Public key for verifying signature
+    @param sigField Object's field containing the signature.
+    If not specified the value defaults to `sfSignature`.
 */
 bool
-verify (STObject const& st,
-    HashPrefix const& prefix,
-        PublicKey const& pk,
-            bool mustBeFullyCanonical);
+verify (STObject const& st, HashPrefix const& prefix,
+    PublicKey const& pk,
+        SF_Blob const& sigField = sfSignature);
```

Comment:
Thanks for getting rid of the `mustBeFullyCanonical` bool.  Those flags always give me the creeps, since I'm never quite sure when they should be `true` or `false`.